loadbalancing.md

```diff
@@ -0,0 +1,345 @@
+
+If you are using Kubernetes you probably want to use the [LoadBalancer](https://docs.pingcap.com/tidb-in-kubernetes/stable/access-tidb#loadbalancer) service type that works with Amazon ELB and similar services from other cloud vendors. For on-premise Kubernetes something like [metallb](https://metallb.universe.tf/) can be used.
+
+When your are not using Kubernetes to deploy on a cloud service you can still use the loadbalancing service from your cloud provider.
+
+Another common option is to use a software based loadbalancer. Common ones are ProxySQL, haproxy and MySQL Router.
+
+The last type of loadbalancer that we will discuss here is hardware based loadbalancers. These are physical machines where you plug in a network cable. These are often costly, but also offer a high throughput.
+
+## General loadbalancer requirements for TiDB
+
+Where loadbalancers may offer advanced features like read/write splitting for MySQL, this is not needed for TiDB. The loadbalancer does not need to be able to inspect the MySQL protocol. A TCP level loadbalacer works fine for TiDB.
+
+The loadbalancer only needs access to the TiDB servers (TCP port 4000 by default) and doesn't need access to PD or TiKV servers. When using the HTTP status access to TCP port 10080 is also required.
+
+It is recommended to use `http://<ip_of_tidb>:10080/status` as the health check since it supports the use of the _graceful shutdown_ feature. This feature allows you to first drain client connections before shutting down a TiDB Server and reduces the impact on applications. The [`graceful-wait-before-shutdown`](/tidb-configuration-file.md#graceful-wait-before-shutdown-new-in-v50) variable needs to be set to a non-zero value to benefit from this feature.
+
+Health checks that check for `read_only` or other MySQL or InnoDB variables should be avoided.
+
+## Where to put your loadbalancer
+
+For software based loadbalancers you have the option to put your loadbalancer on the same hosts as your application or on separate hosts. Both have their own benefits.
+
+Putting the loadbalancer on the same hosts as your application is good for performance as it doesn't need an extra network hop. However having to adminster many instances of your loadbalancing software makes this slightly more complicated to administer.
+
+Putting the loadbalancer on separate machines may be less performant as it needs an extra network hop and depending on your traffic may become a choke-point. However for administration this is might be easier.
+
+Don't forget to include High-Availablility in your decision. With the first setup an unavailable loadbalancer only impacts a single application host. With the second solution you probably have to take some steps to make your loadbalancer software high-available, which complicates the configuration.
```

Comment:
Maybe consider rewriting the above to use ordinal bullet points so the text below makes more sense?

I suggest removing "Don't forget" because it can make the user worry about perils. Both are accepted solutions, so I think we can just say "this should be a factor in your decision".

```suggestion
High availability is also an important consideration when deciding between these options. With the first setup an unavailable loadbalancer only impacts a single application host. With the second solution you probably have to take some steps to make your loadbalancer software high-available, which complicates the configuration.
```